### PR TITLE
Added support for short, short[], byte & byte[]

### PR DESCRIPTION
**Pull request by skigrinder**
## Description

Added support for short, short[], byte & byte[].
Added testing for these types. 

## Motivation and Context
- Improves functionality.
- Closes nanoframework/Home#622.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by skigrinder** [comment]
Added support for short, short[], byte & byte[].
Added testing for these types

**Comment by skigrinder** [comment]
Thanks Jose!

**Comment by josesimoes** [comment]
@skigrinder seems that your working branch is behind develop. Care to rebase please?

**Comment by skigrinder** [comment]
Hi Jose,
I'm reading up on the rebase procedure and don't want to screw things up.
Is it correct to rebase from my current branch (type-additions) onto the 
origin/develop branch?
 From VS, is this what needs to be done or have I misunderstood the 
rebase process?

**Comment by josesimoes** [comment]
Not sure what's your "upstream" repo. It can be "origin" as you have it... The branch it's OK: develop.
Basically you want to pull all the new stuff and commit your changes on top of it.

For the next iteration, make sure you have your develop branch in sync with upstream. Before you branch, that is. 😉

**Comment by skigrinder** [comment]
Hi Jose,
Did the rebase and synced everything into the existing PR.
Sorry for the confusion.  I created this fork shortly before you did a 
bunch of new updates & NuGet releases.
In retrospect, I should have clobbered the fork and created a new one 
but it was good to go through the rebase process.
Hoping I did this correctly.
Can you please take a look and let me know if I need to do anything more?
Thanks,
    Jim

**Comment by josesimoes** [comment]
Yes looks goo now. Thanks!

Now there is a bug reported by sonarcloud. Care to take a look and fix?

**Comment by skigrinder** [comment]
Hi Jose',
Found and fixed the issue identified by sonarcloud.
Thanks,
    Jim

File: source/nanoFramework.Json.sln [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
-EndProject
 Project("{11A8DD76-328B-46DF-9F39-F559912D0360}") = "nanoFramework.Json.Test", "nanoFramework.Json.Test\nanoFramework.Json.Test.nfproj", "{65B0ED92-3D3C-4289-A267-4A4B160FCF13}"
 EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{BAFE1F60-FD27-41B0-9E8C-117F56001A8C}"
-	ProjectSection(SolutionItems) = preProject
-		nanoFramework.Json.nuspec = nanoFramework.Json.nuspec
-		NuGet.Config = NuGet.Config
-		version.json = version.json
-	EndProjectSection
+Project("{11A8DD76-328B-46DF-9F39-F559912D0360}") = "nanoFramework.Json", "nanoFramework.Json\nanoFramework.Json.nfproj", "{181141FB-0D05-41C8-8692-81A2989E6B4B}"
```
Comment (networkfusion) [inline source/nanoFramework.Json.sln:8]: Nuget Spec is required, not sure why it has been deleted...

File: source/nanoFramework.Json.Test/app.config [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
```
Comment (networkfusion) [inline source/nanoFramework.Json.Test/app.config:7]: app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

File: source/nanoFramework.Json.Test/nanoFramework.Json.Test.nfproj [hunk 3]
```diff
@@ -28,17 +28,21 @@
...
+      <SpecificVersion>True</SpecificVersion>
     </Reference>
     <Reference Include="Windows.Storage.Streams, Version=1.10.0.3, Culture=neutral, PublicKeyToken=c07d481e9758c731">
       <HintPath>..\packages\nanoFramework.Windows.Storage.Streams.1.10.0\lib\Windows.Storage.Streams.dll</HintPath>
       <Private>True</Private>
+      <SpecificVersion>True</SpecificVersion>
     </Reference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (networkfusion) [inline source/nanoFramework.Json.Test/nanoFramework.Json.Test.nfproj:45]: app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

File: source/nanoFramework.Json/app.config [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="mscorlib" publicKeyToken="c07d481e9758c731" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.7.3.5" newVersion="1.7.3.5" />
```
Comment (networkfusion) [inline source/nanoFramework.Json/app.config:7]: app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

File: source/nanoFramework.Json/nanoFramework.Json.nfproj [hunk 5]
```diff
@@ -43,21 +43,26 @@
...
+      <SpecificVersion>True</SpecificVersion>
     </Reference>
     <Reference Include="Windows.Storage.Streams, Version=1.10.0.3, Culture=neutral, PublicKeyToken=c07d481e9758c731">
       <HintPath>..\packages\nanoFramework.Windows.Storage.Streams.1.10.0\lib\Windows.Storage.Streams.dll</HintPath>
       <Private>True</Private>
+      <SpecificVersion>True</SpecificVersion>
     </Reference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (networkfusion) [inline source/nanoFramework.Json/nanoFramework.Json.nfproj:65]: app.config files are bad, it shows that there is an issue with the nugets installed. Try removing the file and ensure it still builds after updating the nugets.

**Comment by skigrinder** [comment]
Hi Robin,
Thanks for the help.
I removed the app.config files, cleaned up the extraneous whitespace and 
rebuilt everything.
Please let me know if you see any other issues.
Thanks,
    Jim

**Pull request merged**